apps/web/src/components/tx-flow/flows/SafeAppsTx/index.tsx

```diff
@@ -25,32 +23,34 @@ const SafeAppsTxFlow = ({
   data: SafeAppsTxParams
   onSubmit?: (txId: string, safeTxHash: string) => void
 }) => {
-  const { step, nextStep, prevStep } = useTxStepper(undefined, TxFlowType.SAFE_APPS_TX)
+  const ReviewTransactionComponent = useCallback(
+    (props: ReviewTransactionContentProps) => {
+      return <ReviewSafeAppsTx safeAppsTx={data} {...props} />
+    },
+    [data],
+  )
+
+  const handleSubmit: SubmitCallback = useCallback(
+    (args) => {
+      if (!args || !args.txId) {
+        return
+      }
 
-  const origin = useMemo(() => getTxOrigin(data.app), [data.app])
+      const safeTxHash = args.txId.slice(-66)
```

Comment:
Nit: as mentioned [in your other PR](https://github.com/safe-global/safe-wallet-monorepo/pull/5836#discussion_r2077318064), we might benefit from a helper.

Reply:
I added a small util function